app/helpers/katello/subscription_mailer_helper.rb

```diff
@@ -0,0 +1,18 @@
+module Katello
+  module SubscriptionMailerHelper
+    include Rails.application.routes.url_helpers
+
+    def report_url
+      base_url = report_data_report_template_path(@report_template.id, job_id: @provider_job_id) unless @pools.empty?
+      "#{Setting[:foreman_url]}#{base_url}"
+    end
+
+    def start_report_task(mail_query)
+      @report_template ||= ReportTemplate.find_by(name: "Subscription - Entitlement Report")
```

Comment:
May have misunderstood my original feedback here - but the way this is written now makes the `||=` unnecessary - can just use normal assignment